src/dali/abstract_pair_converter_plugin.py

```diff
@@ -68,9 +68,14 @@ class AssetPairAndTimestamp(NamedTuple):
 
 
 class MappedGraph(Graph[ValueType]):
-    def __init__(self, vertexes: Optional[List["Vertex[ValueType]"]] = None) -> None:
+    def __init__(self, vertexes: Optional[List["Vertex[ValueType]"]] = None, optimized_assets: Optional[Dict[str, None]] = None) -> None:
```

Comment:
I have a few questions:
* I'm not clear why assets are qualified as "optimized": could you clarify the meaning of the term?
* why does init receive an `optimized_assets` parameter? The way I was thinking of it was that init would keep the same signature and clone would receive `changed_edges`, `new_edges` and `deleted_edges`. Would that work or am I missing something?

Reply:
It's probably best illustrated with an example,

I have two transactions for BTC in one month.

The CCXT pair converter goes to price the first transaction, finds it's un-optimized by checking using `is_optimized()` and proceeds to optimize the graph. It makes ~5 REST API calls to find the quote asset with the most volume, calculates new weights and clones the graph. It adds BTC to the `self.__optimized_assets` dictionary when it was cloned. Now the new clone has that asset marked as optimized.

We get to the second transaction that month for BTC. CCXT pair converter checks and finds BTC is optimized and it doesn't have to make the costly REST API calls, saving time.

In an ideal world, we would optimize the entire graph, but of course that would take a prohibitively long time, so we optimize just what is needed and carry that over into the new graph. 

Later, when we pull an ETH transaction, we will optimize that and copy over the optimization into the new graph, so that if we encounter another BTC transaction during that month the optimization is still there for BTC and we don't need to make unnecessary REST API calls.